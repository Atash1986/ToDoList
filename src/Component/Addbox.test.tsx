import { render, screen } from "@testing-library/react";
import Addbox from "./AddBox";
import "@testing-library/jest-dom";

test("containar should be in page", () => {
  render(
    <Addbox
      activeCategoryId={1}
      itemId={1}
      setItemId={() => {}}
      setItems={() => {}}
    />
  );
  const s = screen.getByTestId(/add-box-container/i);
  expect(s).toBeInTheDocument();
});

test("title box should be in page", () => {
  render(
    <Addbox
      activeCategoryId={1}
      itemId={1}
      setItemId={() => {}}
      setItems={() => {}}
    />
  );
  const s = screen.getByTestId(/add-box-title/i);
  expect(s).toBeInTheDocument();
});

test("author box should be in page", () => {
  render(
    <Addbox
      activeCategoryId={1}
      itemId={1}
      setItemId={() => {}}
      setItems={() => {}}
    />
  );

  const s = screen.getByTestId(/add-box-author/i);
  expect(s).toBeInTheDocument();
});

test("add button should be in page", () => {
  render(
    <Addbox
      activeCategoryId={1}
      itemId={1}
      setItemId={() => {}}
      setItems={() => {}}
    />
  );
  const s = screen.getByTestId(/add-box-add-button/i);
  expect(s).toBeInTheDocument();
});

test("error box should be in page", () => {
  render(
    <Addbox
      activeCategoryId={1}
      itemId={1}
      setItemId={() => {}}
      setItems={() => {}}
    />
  );
  const s = screen.getByTestId(/add-box-error-box/i);
  expect(s).toBeInTheDocument();
});
