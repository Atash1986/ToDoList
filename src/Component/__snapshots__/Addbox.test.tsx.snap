// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no items 1`] = `
<div>
  <div
    class="addBoxContainer"
    data-testid="add-box-container"
  >
    <div
      class="addBox"
    >
      <input
        class="taskTitle"
        data-testid="add-box-title"
        data-tooltip-content=""
        data-tooltip-id=""
        name="title"
        type="text"
        value=""
      />
      <select
        data-testid="add-box-author"
        name="Default Value"
      >
        <option
          value="-1"
        >
          Select an author
        </option>
        <option
          id="1"
          value="1"
        >
          Arian
        </option>
        <option
          id="2"
          value="2"
        >
          Mahdiar
        </option>
      </select>
      <button
        class="addButton"
        data-testid="add-box-add-button"
        style="cursor: pointer;"
      >
        <img
          src="plus.svg"
        />
      </button>
      <br />
    </div>
    <div
      class="errorRequirement"
    >
      <span
        data-testid="add-box-error-box"
      />
    </div>
  </div>
</div>
`;
